subsys/bluetooth/mesh/light_temp_srv.c

```diff
@@ -273,7 +273,7 @@ static void lvl_move_set(struct bt_mesh_lvl_srv *lvl_srv,
 	if (move_set->delta != 0 && move_set->transition) {
 		/* Calculate the transition time to edge in the Generic Level space: */
 		int32_t lvl_target = (move_set->delta > 0 ? INT16_MAX : INT16_MIN);
-		int32_t lvl_delta = lvl_target - temp_to_lvl(srv, status.current.temp);
+		int64_t lvl_delta = lvl_target - temp_to_lvl(srv, status.current.temp);
 		uint32_t time_to_edge = (move_set->transition->time * lvl_delta) / move_set->delta;
```

Comment:
```suggestion
		int32_t lvl_delta = lvl_target - temp_to_lvl(srv, status.current.temp);
		uint32_t time_to_edge = ((int64_t)move_set->transition->time * lvl_delta) / move_set->delta;
```

Reply:
I guess it is a matter of taste. I'd prefer to avoid not necessary casting. That makes code awkward.